form/src/Components/Form.js

```diff
@@ -0,0 +1,120 @@
+import "../App.css";
+import React, { useState } from "react";
+import {
+  intro_words,
+  letter_content,
+  farewell,
+} from "../DefaultData/ChristmasPageData";
+
+const Form = ({ storage, firestore, user }) => {
+  const [files, setFiles] = useState([]);
+  const [urls, setUrls] = useState([]);
+  const [openingText, setOpeningText] = useState(intro_words.msg);
+  const [from, setFrom] = useState(user.displayName);
+  const [to, setTo] = useState(intro_words.to);
+  const [letterGreeting, setLetterGreeting] = useState(letter_content.greeting);
+  const [letterText, setLetterText] = useState(letter_content.msg);
+  const [closingText, setClosingText] = useState(farewell.msg);
+  const [signOff, setSignOff] = useState(farewell.from);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const urlTemp = await handleUpload();
+    setUrls([...urlTemp]);
+    handleFireStoreWrite();
+  };
+
+  const handleFireStoreWrite = () => {
+    console.log("Writing to firestore");
+    firestore
+      .collection("SiteData")
+      .add({
+        createdBy: user.email,
+        openingText,
+        from,
+        to,
+        letterGreeting,
+        letterText,
+        closingText,
+        signOff,
+        images: urls,
+      })
+      .then((docRef) => {
+        console.log("Document written with ID: ", docRef.id);
+      })
+      .catch((error) => {
+        console.error("Error adding document: ", error);
+      });
+  };
+
+  const handleUpload = async () => {
+    console.log("Uploading Files to storage bucket...");
+    const urltemp = await [...files].map(async (file) => {
+      const snapshot = await storage
```

Comment:
i think here too... we can remove the await and the console log in a `.then()`

Reply:
Yea we probably don't need that but I just didn't wanna mix `.then()` and `await`. Generally, I find `await` to be much more readable because it follows more of a traditional assignment syntax.